ykcompile/src/lib.rs

```diff
@@ -1730,4 +1740,33 @@ mod tests {
         ct.execute(&mut args);
         assert_eq!(args.0, 6);
     }
+
+    #[do_not_trace]
```

Comment:
I wonder if we should rename the attribute so that it is more obvious it's part of Yorick.

Rustc already has an attribute mentioning tracing, which might be confusing: `trace_caller`.

Something like `yk_notrace`?

Reply:
If you think it's necessary to make this clear, fine by me. Although, since people need to specifically download our version of rustc to get tracing, there shouldn't be much confusion.

Reply:
It's a long shot, but there may be a future where this stuff is merged into upstream rustc. It might be an idea to change the name now, before we have users.

Reply:
Ok. I will have to update the companion PR as well.

Reply:
Should we also rename `trace_tail` to `yk_trace_tail` while we are here? In think there's even more confusion here, since `trace_caller` and `trace_tail` look quite similar but have nothing to do with each other.